server/common/fbs/matrix.py

```diff
@@ -46,74 +50,91 @@ def serialize_matrix(builder, n_rows, n_cols, columns, col_idx):
 # Serialization helper
 def serialize_typed_array(builder, source_array, encoding_info):
     """
-    Serialize any of the various typed arrays, eg, Float32Array. Specific means of serialization and type conversion
+    Serialize any of the various typed arrays, eg, Float32FBArray. Specific means of serialization and type conversion
     are provided by type_info.
     """
 
     arr = source_array
-    (array_type, as_type) = encoding_info(source_array)
-
+    (array_type, as_type1, as_type2) = encoding_info(source_array)
     if isinstance(arr, pd.Index):
         arr = arr.to_series()
 
-    # convert to a simple ndarray
-    if as_type == "json":
+    types = [as_type1, as_type2, "int"]
+
+    if as_type1 == "json":
         as_json = arr.to_json(orient="records")
         arr = np.array(bytearray(as_json, "utf-8"))
+        fields = [arr]
     else:
         if sparse.issparse(arr):
-            arr = arr.toarray()
+            row = arr.row
+            data = arr.data
+            size = np.array([max(arr.shape)], dtype="int")
```

Comment:
the FB schema defines this field as an int32, but here you are creating an int64. Quite possibly a bug, or you are later doing an unnecessary cast to int32.  Either way, shouldn't this read `dtype="int32"`?

Reply:
Thanks for the catch - will fix.